addons/account/models/account_move_send.py

```diff
@@ -651,16 +651,28 @@ def _generate_invoice_fallback_documents(self, invoices_data):
                     .create(invoice_data.pop('proforma_pdf_attachment_values'))
 
     def _check_sending_data(self, moves, **custom_settings):
-        """Assert the data provided to _generate_and_send_invoices are correct.
+        """Check the data provided to _generate_and_send_invoices are correct.
         This is a security in case the method is called directly without going through the wizards.
         """
         self._check_move_constrains(moves)
-        assert all(self._get_default_pdf_report_id(move).is_invoice_report for move in moves)
-        assert custom_settings['pdf_report'].is_invoice_report if custom_settings.get('pdf_report') else True
-        assert all(
-            sending_method in dict(self.env['res.partner']._fields['invoice_sending_method'].selection)
-            for sending_method in custom_settings.get('sending_methods', [])
-        ) if 'sending_methods' in custom_settings else True
```

Comment:
Imo this assert can stay an assert, it's not supposed to be possible to do through UI, it's a safeguard.
Then I'd just add a `_check_invoice_report` call that does your raise, instead of the two asserts.